pkg/client/options.go

```diff
@@ -187,6 +220,25 @@ type CreateOptions struct {
 	// this request.  It must be set with server-side apply.
 	FieldManager string
 
+	// fieldValidation instructs the server on how to handle
+	// objects in the request (POST/PUT/PATCH) containing unknown
+	// or duplicate fields. Valid values are:
+	// - Ignore: This will ignore any unknown fields that are silently
+	// dropped from the object, and will ignore all but the last duplicate
+	// field that the decoder encounters. This is the default behavior
+	// prior to v1.23.
+	// - Warn: This will send a warning via the standard warning response
+	// header for each unknown field that is dropped from the object, and
+	// for each duplicate field that is encountered. The request will
+	// still succeed if there are no other errors, and will only persist
+	// the last of any duplicate fields. This is the default in v1.23+
+	// - Strict: This will fail the request with a BadRequest error if
+	// any unknown fields would be dropped from the object, or if any
+	// duplicate fields are present. The error returned from the server
+	// will contain all unknown and duplicate fields encountered.
+	// +optional
```

Comment:
This seems like one of the very few times that we use the optional marker here. 

Not sure what we want to do, but consistency would be probably (if we have optional only on some fields this would imply the others are required?)

Reply:
Just a consequence of copy-paste. I'll remove the marker.

Reply:
Done in https://github.com/kubernetes-sigs/controller-runtime/pull/2860/commits/725259f79a67a0bcce6f694ee3e50a26a352884b